src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import { Swiper, SwiperSlide } from "swiper/react";
+import { ScreenBreakPoints } from '../../constants/BreakPoints';
+import Stars from '../stars/stars';
+import IconMeaning from '../IconMeaning/IconMeaning';
+import Price from '../Price/Price';
+import {
+    CarouselWrapper,
+    CardWrapper,
+    Title,
+    Description
+} from './styles';
+
+interface CardData {
+    id: number;
+    title: string;
+    imageUrl: string;
+    description?: string;
+    icon?: 'spicy' | 'vegetarian' | 'vegan';
+    chefname?: string;
+    stars?: number;
+    price?: number;
+}
+
+interface CarouselProps {
+    cards: CardData[];
+}
+
+const Carousel: React.FC<CarouselProps> = ({ cards }) => {
+    const [isDesktop, setIsDesktop] = useState(false);
+
+    useEffect(() => {
+        const handleResize = () => {
+            const isLargeScreen = window.innerWidth >= ScreenBreakPoints.lg;
+            setIsDesktop(isLargeScreen);
+        };
+
+        handleResize();
+
+        window.addEventListener('resize', handleResize);
+
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
+    }, []);
+
+    return (
+        <CarouselWrapper>
+            <Swiper
+                spaceBetween={10}
+                slidesPerView={isDesktop ? 3 : 1.1}
+                loop={!isDesktop ? true : false}
+                allowTouchMove={!isDesktop}
```

Comment:
im pretty sure u can set all these in the break points below.
if not, it might be a better idea to make a custom hook that handles the resize instead of doing it in the component itself - since u will be able to reuse it in other components if and when needed.

Reply:
I fixed it on Thursday. There's an updated version of the file in the frontend branch. I have PR too, from branch: frontend to dev.